plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3ClientSettings.java

```diff
@@ -401,27 +400,64 @@ private static S3BasicCredentials loadCredentials(Settings settings, String clie
-                getConfigValue(settings, clientName, PROXY_PORT_SETTING),
-                proxyUsername.toString(),
-                proxyPassword.toString(),
-                Math.toIntExact(getConfigValue(settings, clientName, READ_TIMEOUT_SETTING).millis()),
-                getConfigValue(settings, clientName, MAX_RETRIES_SETTING),
-                getConfigValue(settings, clientName, USE_THROTTLE_RETRIES_SETTING),
-                getConfigValue(settings, clientName, USE_PATH_STYLE_ACCESS),
-                getConfigValue(settings, clientName, DISABLE_CHUNKED_ENCODING),
-                getConfigValue(settings, clientName, REGION),
-                getConfigValue(settings, clientName, SIGNER_OVERRIDE)
+        final Protocol awsProtocol = getConfigValue(settings, clientName, PROTOCOL_SETTING);
+        return new S3ClientSettings(
+            S3ClientSettings.loadCredentials(settings, clientName),
+            getConfigValue(settings, clientName, ENDPOINT_SETTING),
+            awsProtocol,
+            Math.toIntExact(getConfigValue(settings, clientName, READ_TIMEOUT_SETTING).millis()),
+            getConfigValue(settings, clientName, MAX_RETRIES_SETTING),
+            getConfigValue(settings, clientName, USE_THROTTLE_RETRIES_SETTING),
+            getConfigValue(settings, clientName, USE_PATH_STYLE_ACCESS),
+            getConfigValue(settings, clientName, DISABLE_CHUNKED_ENCODING),
+            getConfigValue(settings, clientName, REGION),
+            getConfigValue(settings, clientName, SIGNER_OVERRIDE),
+            validateAndCreateProxySettings(settings, clientName, awsProtocol)
+        );
+    }
+
+    static ProxySettings validateAndCreateProxySettings(final Settings settings, final String clientName, final Protocol awsProtocol) {
+        ProxySettings.ProxyType proxyType = getConfigValue(settings, clientName, PROXY_TYPE_SETTING);
+        final String proxyHost = getConfigValue(settings, clientName, PROXY_HOST_SETTING);
+        final int proxyPort = getConfigValue(settings, clientName, PROXY_PORT_SETTING);
+        final SecureString proxyUserName = getConfigValue(settings, clientName, PROXY_USERNAME_SETTING);
+        final SecureString proxyPassword = getConfigValue(settings, clientName, PROXY_PASSWORD_SETTING);
+        if (awsProtocol != Protocol.HTTPS && proxyType == ProxySettings.ProxyType.DIRECT && proxyPort != 0 && Strings.hasText(proxyHost)) {
+            // This is backward compatibility for the current behaviour.
+            // The default value for Protocol settings is HTTPS,
+            // The expectation of ex-developers that protocol is the same as the proxy protocol
+            // which is a separate setting for AWS SDK.
+            // In this case, proxy type should be the same as a protocol,
+            // when proxy host and port have been set
+            proxyType = ProxySettings.ProxyType.valueOf(awsProtocol.name());
+            LOGGER.warn(
```

Comment:
You may consider switching to  `DeprecationLogger` instead: 
```
private static final DeprecationLogger LOGGER = DeprecationLogger.getLogger(S3ClientSettings.class);
```

Reply:
good point